api-gateway/src/main/resources/static/scripts/inventory-list/inventory-list.controller.js

```diff
@@ -1,40 +1,58 @@
 'use strict';
 
 angular.module('inventoryList')
-    .controller('InventoryListController', ['$http', '$scope', function ($http, $scope) {
+    .controller('InventoryListController', ['$http', '$scope', '$stateParams', '$state', function ($http, $scope, $stateParams, $state) {
         var self = this;
+        self.currentPage = $stateParams.page = 0
+        self.listSize = $stateParams.size = 10
+        self.realPage = parseInt(self.currentPage) + 1
+        var numberOfPage
+        var name
+        var type
+        var desc
+
+        getInventoryList()
 
-                $http.get('api/gateway/inventory').then(function (resp) {
-                    self.inventoryList = resp.data;
-                    console.log("Resp data: " + resp.data)
-                    console.log("inventory list: " + self.inventoryList)
-                });
-//search by inventory field
         $scope.searchInventory = function (inventoryName, inventoryType, inventoryDescription){
+            getInventoryList(inventoryName, inventoryType, inventoryDescription)
+        }
+//search by inventory field
+        function getInventoryList(inventoryName, inventoryType, inventoryDescription){
```

Comment:
Great work Emile, do you mind explaining what this line of code does

Reply:
Because i can't call a scope function inside the controller. So to get it from the search and use it inside the controller. If you have any other way tell me.

Reply:
I understand, what you did was good, good jib